sandpiper/processes/wps_parser.py

```diff
@@ -89,18 +80,27 @@ def _handler(self, request, response):
             log_level=loglevel,
             process_step="process",
         )
+        parsed_vars = {}
 
-        try:
-            parse_tree, vars, region_var = build_parse_tree(condition)
-        except SyntaxError as e:
-            raise ProcessError(f"{type(e).__name__}: Invalid syntax in condition")
-        except ValueError as e:
-            raise ProcessError(
-                f"{type(e).__name__}: variable name should have 5 values, variable, "
-                "time_of_year, temporal, spatial, and percentile"
-            )
-        except Exception as e:
-            raise ProcessError(f"{type(e).__name__}: {e}")
+        for condition in conditions:
+            try:
+                parse_tree, vars, region_var = build_parse_tree(condition.data)
+                parsed_vars[condition.data] = {
+                    "parse_tree": parse_tree,
+                    "variables": vars,
+                    "region_variable": region_var,
+                }
```

Comment:
Try using dict comprehension again! Might be a bit more complex here though.

Reply:
Tried a bunch of stuff for dictionary comprehension and couldn't find anything that worked.